pallet/account-migration/src/lib.rs

```diff
@@ -284,25 +283,37 @@ pub mod pallet {
 }
 pub use pallet::*;
 
+const SIGN_TIPS: &[u8] = b"\
+	Hi there from `account-migration`! \
+	Sign this message to prove you have access to this wallet and we'll do the migration for you. \
+	This won't cost you any. \
+	To stop hackers using your wallet, here's a unique message ID: \
+";
+
```

Comment:
Why not add this on the UI side? I think this is should not be part of the protocol. It's not friendly to the UI developers to copy this `TIP` there, easy to make mistakes.

@hackfisher What do you think?

Reply:
The message is verified on chain.

Reply:
the UI needs to follow this message pattern to create a signature, right? Otherwise, the signature they created can not be verified on the chain.

Reply:
Yep. Just imagine a pallet is a low-level DApp.

Reply:
Just because pallet is a low-level app, it should be kept simple. The tip should be left to a higher level application and the pallet should focus on the validation.

Reply:
I mean. We should treat this as a Dapp. This is its job.

We add this to tell the user what things they are signing.

Then, what's your suggestion? How to verify the signature if we only add this on frontend?

Reply:
> We add this to tell the user what things they are signing.

The purpose is to prompt the user to understand what they are doing before signing, but it is not necessary to include the tips in the signature. Imagine signing an ethereum or substrate with no tip and only a transaction structure.

> what's your suggestion?

No need to do this. `hash(chain_id, spec_name, b"xx", account_id)` is enough for on-chain verification.

> How to verify the signature if we only add this on frontend?

In my opinion, we don't need to verify the content of the tip.

Reply:
But users don't know what they are signing.

There is no reason for users to trust your tip on the website. What if you put a encoded  extrinsic in the message field? They don't know it at all.

`hash(chain_id, spec_name, b"xx", account_id)` is just some bytes.

This is what good DApps did.

<img width="343" alt="image" src="https://user-images.githubusercontent.com/16152305/211714652-f83092a2-5ec3-4891-9b18-ecb656954d70.png">

Reply:
Yes, exactly, these changes are made on my request based on the same reason.

Users need to know what message they are sign, since the message is designable, then human readable is better.

Redesign of the message format does not make the protocol more complicate.